src/derivatives.jl

```diff
@@ -46,14 +53,20 @@ abstract type Swaption <: Derivative end
 - `θ`: long term mean level
 - `t`: time interval
 """
-@kwdef struct InterestRateDerivative{KT<:Real,RT<:Real,σT<:Real,θT<:Real,TT<:Real}
+struct InterestRateDerivative{KT<:Real,RT<:Real,σT<:Real,θT<:Real,TT<:Real}
     k::KT
     r::RT
     σ::σT
     θ::θT
     t::TT
 end
 
+"Construct `InterestRateDerivative` with parameters and start `t0` and final `t1` `Date`s."
+function InterestRateDerivative(; k::Real, r::Real, σ::Real, θ::Real, t0::Date, t1::Date)
+    t = time_to_expiration(t0, t1)
```

Comment:
perhaps consider adding a fool-proof check for t1 > t0 ?

Reply:
Sure, makes sense. Will add throw an `ArgumentError` in that case.